adapters/compass/compass.go

```diff
@@ -0,0 +1,147 @@
+package compass
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+	var err error
+	var adapterRequests []*adapters.RequestData
+
+	reqCopy := *request
+	for _, imp := range request.Imp {
+		reqCopy.Imp = []openrtb2.Imp{imp}
+
+		var bidderExt adapters.ExtImpBidder
+		var compassExt openrtb_ext.ImpExtCompass
+
+		if err = json.Unmarshal(reqCopy.Imp[0].Ext, &bidderExt); err != nil {
+			return nil, append(errs, err)
+		}
+		if err = json.Unmarshal(bidderExt.Bidder, &compassExt); err != nil {
+			return nil, append(errs, err)
+		}
+
+		finalyImpExt := reqCopy.Imp[0].Ext
+		if compassExt.PlacementID != "" {
+			finalyImpExt, _ = json.Marshal(map[string]interface{}{
+				"bidder": map[string]interface{}{
+					"placementId": compassExt.PlacementID,
+					"type":        "publisher",
+				},
+			})
+		} else if compassExt.EndpointID != "" {
```

Comment:
From the logic of this `if/else if` statement it seems that we are expecting that either `"placementId"` or `"endpointId"` are non-empty strings but the following two corner cases added to `adapters/compass/params_test.go` in my local branch seem to be able to pass JSON schema validation and would not end in either branch of the aforementioned `if/else if` statement. Test cases shown below:
```
36   var validParams = []string{
37       `{"placementId": "test"}`,
38       `{"placementId": "1"}`,
39       `{"endpointId": "test"}`,
40       `{"endpointId": "1"}`,
   +     `{"endpointId": ""}`,
   +     `{"placementId": ""}`,
41   }
adapters/compass/params_test.go
```
In case these scenarios are not intended to skip the branches of the `if/else` statment, we can prevent them from passing validation by adding a `minLength` fields in the JSON schema file `static/bidder-params/compass.json`:
```
 1   {
 2       "$schema": "http://json-schema.org/draft-04/schema#",
 3       "title": "Compass Adapter Params",
 4       "description": "A schema which validates params accepted by the Compass adapter",
 5       "type": "object",
 6  
 7       "properties": {
 8         "placementId": {
 9           "type": "string",
   +         "minLength": 1,
10           "description": "Placement ID"
11         },
12         "endpointId": {
13           "type": "string",
   +         "minLength": 1,
14           "description": "Endpoint ID"
15         }
16       },
17       "oneOf": [
18         { "required": ["placementId"] },
19         { "required": ["endpointId"] }
20       ]
21     }
static/bidder-params/compass.json
```
Do you believe these `"minLength"` fields should be added?